src/components/ActionNameCard.tsx

```diff
@@ -50,10 +53,14 @@ const ActionNameCard = ({
 
   const debouncedSetActionName = useMemo(
     () =>
-      debounce((id: ActionData["ID"], name: string) => {
-        setActionName(id, name);
-      }, 400),
-    [setActionName]
+      debounce(
+        (id: ActionData["ID"], name: string) => {
+          setActionName(id, name);
+        },
+        400,
+        { leading: localName.length === 0 }
```

Comment:
Added leading option to debounce as suggested on the call.

Reply:
This just needs to be `leading: true` without the localName dep. Otherwise the debouncing function is changing (we're calling different functions) and it doesn't work as expected.

Reply:
Changed here --> [7ab7c4f](https://github.com/microbit-foundation/ml-trainer/pull/553/commits/7ab7c4ff9cea38f98959ba27c2c3c1407bab45b8)